elements/lisk-codec/src/bytes.ts

```diff
@@ -11,14 +11,12 @@
  *
  * Removal or modification of this copyright notice is prohibited.
  */
-import { writeVarInt } from './varint';
+import { writeUInt32, readUInt32 } from './varint';
 
-interface SchemaProperty {
-	readonly dataType: string;
-}
-
-export const writeBytes = (bytes: Buffer, _schema: SchemaProperty): Buffer =>
-	Buffer.concat([writeVarInt(bytes.length, { dataType: 'uint32' }), bytes]);
+export const writeBytes = (bytes: Buffer): Buffer =>
+	Buffer.concat([writeUInt32(bytes.length), bytes]);
 
-export const readBytes = (buffer: Buffer, _schema: SchemaProperty): Buffer =>
-	buffer.slice(1, buffer.length);
+export const readBytes = (buffer: Buffer, offset: number): [Buffer, number] => {
+	const [byteLength, keySize] = readUInt32(buffer, offset);
```

Comment:
why you read all the bytes as varints? bytes are encoded as bytes and no transformation should happen to them

Reply:
bytes is wire type 2, so the first X bytes are the length of the bytes that comes after.
Therefore, it needs to read the varint first, then take out the bytes that was defined in the bytes

Reply:
Oh so then this is both reading a chunk and also decoding the key? If that's the case maybe we should rename the "readers" to something more meaningful as they will no longer be reading (extracting) a chunk from the main buffer but instead they would be decoding key and slicing the buffer.

Reply:
maybe we could add some comments on top of this read function as the name itself is a bit abstract and hard to follow whats going on inside